reporters.py

```diff
@@ -113,37 +130,137 @@ def print(self):
+    print(f'    report_length={report_length}, alpha={self.alpha}, compare_type={self.compare_type}')
+    if self.label_files is not None:
+      print(self.label_files)
+
+    print(f'--- {report_length} n-grams that System 1 had higher {self.compare_type}')
+    for k, v in self.scorelist[:report_length]:
+      print('{}\t{} (sys1={}, sys2={})'.format(' '.join(k), v, self.matches1[k], self.matches2[k]))
+    print(f'\n--- {report_length} n-grams that System 2 had higher {self.compare_type}')
+    for k, v in reversed(self.scorelist[-report_length:]):
+      print('{}\t{} (sys1={}, sys2={})'.format(' '.join(k), v, self.matches1[k], self.matches2[k]))
+    print()
+
+  def plot(self, output_directory='outputs', output_fig_file='score', output_fig_format='pdf'):
     pass 
-  def plot(self):
+
+  def write_html(self, output_directory='outputs', output_html_file='score.html'):
     pass 
 
 class SentenceReport(Report):
+  def __init__(self, bucketer=None, bucket_type=None, sys1_stats=None, sys2_stats=None, statistic_type=None, score_measure=None):
+    self.bucketer = bucketer
+    self.bucket_type = bucket_type 
+    self.sys1_stats = [s for s in sys1_stats]
+    self.sys2_stats = [s for s in sys2_stats] 
+    self.statistic_type = statistic_type
+    self.score_measure = score_measure
+
   def print(self):
-    bucketer, stats1, stats2, bucket_type, statistic_type, score_measure = self.bucketer, self.stats1, self.stats2, self.bucket_type, self.statistic_type, self.score_measure
+    bucketer, stats1, stats2, bucket_type, statistic_type, score_measure = self.bucketer, self.sys1_stats, self.sys2_stats, self.bucket_type, self.statistic_type, self.score_measure
     self.print_header('Sentence Bucket Analysis')
     print(f'--- bucket_type={bucket_type}, statistic_type={statistic_type}, score_measure={score_measure}')
     for bs, s1, s2 in zip(bucketer.bucket_strs, stats1, stats2):
       print(f'{bs}\t{s1}\t{s2}')
     print()
 
-  def plot(self):
-    pass 
+  def plot(self, output_directory='outputs', output_fig_file='word-acc', output_fig_format='pdf'):
+    width = 0.35
+    sys1 = self.sys1_stats
+    sys2 = self.sys2_stats
+    xlabel = [s for s in self.bucketer.bucket_strs] 
+
+    fig, ax = plt.subplots()
+    ind = np.arange(len(sys1))
+    p1 = ax.bar(ind, sys1, width, color='r', bottom=0)
+    p2 = ax.bar(ind+width, sys2, width, color='#f4e604', bottom=0)
+    ax.set_xticks(ind + width / 2)
+    ax.set_xticklabels(xlabel)
+    plt.xticks(rotation=45)
+    ax.legend((p1[0], p2[0]), ('Sys1', 'Sys2'))
+    ax.autoscale_view()
+
+    if not os.path.exists(output_directory):
+      os.makedirs(output_directory)
+    plt.savefig(os.path.join(output_directory, f'{output_fig_file}.{output_fig_format}'), 
+                format=output_fig_format, bbox_inches='tight')
+
+  def write_html(self, output_directory='outputs', output_html_file='word-acc.html'):
+    pass
 
 def create_reporter_from_profile(profile,
```

Comment:
It's not clear to me why this function is necessary... Couldn't we just call the constructors directly?

Reply:
Yes. I think it is not necessary. `create_reporter_from_profile` was previously used to create a subclass of `Reporter` by passing a dictionary. In the current version, I change the interface by explicitly passing the arguments instead of a dictionary. I will call the constructors instead.